### ISSUE TITLE
infinite uploading metadata when  s3qlctrl upload-meta

### ISSUE THREAD

**Issue by segator** [s3ql/s3ql#179]
**Version:** Latest commit in master branch
**Issue:** After execute 
```bash
s3qlctrl upload-meta /mnt/s3ql/
```
  I notice that is trying to upload metadata forever
This is what I have in the mount.log
```
2020-06-19 08:39:05.223 2124498:MainThread s3ql.mount.get_metadata: Using cached metadata.
2020-06-19 08:39:05.263 2124498:MainThread s3ql.mount.main_async: Creating NFS indices...
2020-06-19 08:39:05.385 2124498:MainThread s3ql.block_cache.__init__: Loaded 10210 entries from cache
2020-06-19 08:39:05.385 2124498:MainThread s3ql.mount.main_async: Mounting s3://test at /mnt/s3ql...
2020-06-19 08:39:05.409 2124703:MainThread s3ql.daemonize.detach_process_context: Daemonizing, new PID is 2124704
2020-06-19 11:09:03.519 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:09:03.560 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:09:03.902 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:09:05.134 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:09:05.511 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:09:09.242 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:09:09.243 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:09:09.286 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:09:09.414 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:09:13.209 2124704:trio-worker-3005 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:09:22.823 2124704:trio-worker-3005 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:09:22.824 2124704:trio-worker-3005 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:09:22.824 2124704:trio-worker-3005 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:09:41.578 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:09:41.579 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:09:41.828 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:09:42.335 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:09:42.551 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:09:44.240 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:09:44.241 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:09:44.257 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:09:44.344 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:09:45.889 2124704:trio-worker-3019 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:09:55.073 2124704:trio-worker-3019 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:09:55.073 2124704:trio-worker-3019 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:09:55.073 2124704:trio-worker-3019 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:10:15.775 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:10:15.776 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:10:16.073 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:10:16.576 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:10:16.810 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:10:18.563 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:10:18.563 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:10:18.580 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:10:18.677 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:10:20.568 2124704:trio-worker-3030 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:10:30.638 2124704:trio-worker-3030 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:10:30.638 2124704:trio-worker-3030 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:10:30.638 2124704:trio-worker-3030 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:10:48.692 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:10:48.692 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:10:48.946 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:10:49.404 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:10:49.644 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:10:51.382 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:10:51.383 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:10:51.398 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:10:51.478 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:10:53.248 2124704:trio-worker-3044 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:11:02.978 2124704:trio-worker-3044 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:11:02.978 2124704:trio-worker-3044 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:11:02.978 2124704:trio-worker-3044 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:11:21.099 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:11:21.099 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:11:21.366 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:11:21.806 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:11:22.021 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:11:23.719 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:11:23.719 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:11:23.735 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:11:23.816 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:11:25.418 2124704:trio-worker-3055 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:11:34.758 2124704:trio-worker-3055 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:11:34.758 2124704:trio-worker-3055 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:11:34.758 2124704:trio-worker-3055 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:11:53.030 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:11:53.030 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:11:53.278 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:11:53.711 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:11:53.928 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:11:55.603 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:11:55.603 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:11:55.618 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:11:55.697 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:11:57.551 2124704:trio-worker-3065 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:12:07.175 2124704:trio-worker-3065 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:12:07.175 2124704:trio-worker-3065 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:12:07.175 2124704:trio-worker-3065 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:12:24.840 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:12:24.840 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:12:25.108 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:12:25.599 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:12:25.814 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:12:27.502 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:12:27.502 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:12:27.517 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:12:27.597 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:12:29.343 2124704:trio-worker-3076 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:12:38.953 2124704:trio-worker-3076 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:12:38.953 2124704:trio-worker-3076 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:12:38.954 2124704:trio-worker-3076 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:12:58.011 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:12:58.012 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:12:58.276 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:12:58.722 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:12:58.942 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:13:00.641 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:13:00.642 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:13:00.657 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:13:00.738 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:13:02.397 2124704:trio-worker-3089 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:13:12.176 2124704:trio-worker-3089 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:13:12.176 2124704:trio-worker-3089 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:13:12.176 2124704:trio-worker-3089 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:13:30.818 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:13:30.819 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:13:31.083 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:13:31.530 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:13:31.745 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:13:33.423 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:13:33.424 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:13:33.439 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:13:33.519 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:13:35.172 2124704:trio-worker-3102 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:13:44.756 2124704:trio-worker-3102 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:13:44.756 2124704:trio-worker-3102 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:13:44.756 2124704:trio-worker-3102 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:14:02.955 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:14:02.955 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:14:03.210 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:14:03.682 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:14:03.939 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:14:05.661 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:14:05.661 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:14:05.677 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:14:05.755 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:14:07.396 2124704:trio-worker-3112 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:14:16.787 2124704:trio-worker-3112 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:14:16.788 2124704:trio-worker-3112 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:14:16.788 2124704:trio-worker-3112 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:14:35.835 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:14:35.835 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:14:36.157 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:14:36.813 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:14:37.032 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:14:38.730 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:14:38.731 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:14:38.759 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:14:38.839 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:14:40.542 2124704:trio-worker-3127 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:14:50.506 2124704:trio-worker-3127 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:14:50.506 2124704:trio-worker-3127 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:14:50.506 2124704:trio-worker-3127 s3ql.metadata.cycle_metadata: Backing up old metadata...
2020-06-19 11:15:12.977 2124704:MainThread s3ql.mount.run: Dumping metadata...
2020-06-19 11:15:12.977 2124704:MainThread s3ql.metadata.dump_metadata: ..objects..
2020-06-19 11:15:13.234 2124704:MainThread s3ql.metadata.dump_metadata: ..blocks..
2020-06-19 11:15:13.717 2124704:MainThread s3ql.metadata.dump_metadata: ..inodes..
2020-06-19 11:15:13.931 2124704:MainThread s3ql.metadata.dump_metadata: ..inode_blocks..
2020-06-19 11:15:15.619 2124704:MainThread s3ql.metadata.dump_metadata: ..symlink_targets..
2020-06-19 11:15:15.619 2124704:MainThread s3ql.metadata.dump_metadata: ..names..
2020-06-19 11:15:15.660 2124704:MainThread s3ql.metadata.dump_metadata: ..contents..
2020-06-19 11:15:15.747 2124704:MainThread s3ql.metadata.dump_metadata: ..ext_attributes..
2020-06-19 11:15:17.388 2124704:trio-worker-3140 s3ql.metadata.upload_metadata: Compressing and uploading metadata...
2020-06-19 11:15:27.727 2124704:trio-worker-3140 s3ql.metadata.upload_metadata: Wrote 31.8 MiB of compressed metadata.
2020-06-19 11:15:27.727 2124704:trio-worker-3140 s3ql.metadata.upload_metadata: Cycling metadata backups...
2020-06-19 11:15:27.727 2124704:trio-worker-3140 s3ql.metadata.cycle_metadata: Backing up old metadata...
```

**Comment by Nikratio** [comment]
Thanks for the report! Could you upgrade the `trio` package to the newest version and test if that fixes the issue?

**Comment by segator** [comment]
I have the 0.15.0,  not sure how to install a newest one

**Comment by segator** [comment]
why is this closed? are you sure that upgrading trio will fix the issue? and to which version?

**Comment by powerpaul17** [comment]
Because @Nikratio fixed the error apparently in commit [65ee19a](https://github.com/s3ql/s3ql/commit/65ee19a3bfa4dd257c05ab94a639b4f1d4576603).

**Comment by segator** [comment]
awesome, but then trio with 0.15 is ok
